### DISCUSSION TITLE
Issue with Data Processing after Updating to Seurat 5.0.1

### DISCUSSION THREAD

**Discussion by VishD17** [satijalab/seurat#8122] [Q&A]
Hello,

Recently I have been getting an error, despite the code working in the past. The input data was not changed in any way and I noticed this error was occurring only after updating to Seurat 5.0.1. In addition, when running on a computer with Seurat 4.3.0 and Seurat Object 4.1.3, the code runs just fine. Below is the error:

#Converting from h5ad to h5seurat
Convert("Mouse_PR_clean.h5ad", dest = "h5seurat", overwrite = TRUE)

#Loading the h5seurat dataset
mouse <- LoadH5Seurat("Mouse_PR_clean.h5seurat", meta.data = FALSE, misc = FALSE, array = "RNA")

#Standard Pre-processing of Mouse Dataset
mouse <- NormalizeData(mouse)
mouse <- FindVariableFeatures(mouse, nfeatures = 2000)

Error in order(hvf.info$vst.variance.standardized, decreasing = TRUE) :
argument 1 is not a vector

To fix this error, I have been trying to degrade the Seurat version following the instructions on the seurat website, and although I converted Seurat and Seurat Object to the required lower versions, the error still persists, which may be because the CRAN dependency packages are still a higher version. Is this a correct interpretation of the error and how is it recommended to fix it? Thanks in advance!

**Comment by saketkc** [comment]
If you have tried this with Seurat4, can you ensure `packageVersion("Seurat")` and `packageVersion("SeuratObject")` both return <5.0. It is always a good idea to restart your rsession after a package downgrade/upgrade.

I cannot reproduce this at my end. Can you try:

```r
mouse <- LoadH5Seurat("Mouse_PR_clean.h5seurat", meta.data = FALSE, misc = FALSE, array = "RNA")
mouse <- UpdateSeuratObject(mouse)
mouse <- NormalizeData(mouse)
mouse <- FindVariableFeatures(mouse, nfeatures = 2000)
```

and see if that fixes it? Also can you please report output of `sessionInfo()` if you run into issues?

**Reply by VishD17** [reply]
Hi @saketkc thanks for sharing!

Running ```mouse <- UpdateSeuratObject(mouse)``` allowed me to load in the dataset and run the FindVariableFeatures. However, the pipeline is now getting stuck at RunUMAP:
```
mouse <- RunUMAP(mouse, dims = 1:50)
10:59:22 UMAP embedding parameters a = 0.9922 b = 1.112
10:59:22 Read 1521 rows and found 50 numeric columns
10:59:22 Using Annoy for neighbor search, n_neighbors = 30
10:59:22 Building Annoy index with metric = cosine, n_trees = 50
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
10:59:22 Writing NN index file to temp file /var/folders/jr/jwtk4wcd5w3d6nx2cthjktpw0000gn/T//RtmpZZAIlF/file46805dee6f3d
10:59:22 Searching Annoy index using 1 thread, search_k = 3000
10:59:22 Annoy recall = 100%
10:59:22 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
10:59:23 Initializing from normalized Laplacian + noise (using irlba)
Error in irlba::irlba(L, nv = n, nu = 0, maxit = iters) : 
  function 'as_cholmod_sparse' not provided by package 'Matrix'
```
I believe this error discussed here (https://github.com/bwlewis/irlba/issues/70) and the suggested fix was to downgrading the Matrix package to 1.6-1. 
```
url <- "https://cran.r-project.org/src/contrib/Archive/Matrix/Matrix_1.6-1.tar.gz"
install.packages(url, repos = NULL, Source = TRUE)
```

The error that I get when running these lines:

```Warning in install.packages: installation of package ‘/var/folders/jr/jwtk4wcd5w3d6nx2cthjktpw0000gn/T//RtmpZZAIlF/downloaded_packages/Matrix_1.6-1.tar.gz’ had non-zero exit status```


This is my current sessionInfo()

```R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

loaded via a namespace (and not attached):
 [1] utf8_1.2.4          future_1.33.0       generics_0.1.3     
 [4] KernSmooth_2.23-22  lattice_0.21-8      listenv_0.9.0      
 [7] digest_0.6.33       magrittr_2.0.3      grid_4.3.2         
[10] RColorBrewer_1.1-3  fastmap_1.1.1       Matrix_1.6-4       
[13] ggrepel_0.9.4       SeuratObject_4.1.3  survival_3.5-7     
[16] httr_1.4.7          fansi_1.0.5         scales_1.3.0       
[19] ica_1.0-3           codetools_0.2-19    cli_3.6.1          
[22] rlang_1.1.2         parallelly_1.36.0   RcppAnnoy_0.0.21   
[25] future.apply_1.11.0 splines_4.3.2       cowplot_1.1.1      
[28] munsell_0.5.0       RANN_2.6.1          Rtsne_0.16         
[31] tools_4.3.2         parallel_4.3.2      dplyr_1.1.4        
[34] colorspace_2.1-0    ggplot2_3.4.4       globals_0.16.2     
[37] ROCR_1.0-11         fitdistrplus_1.1-11 vctrs_0.6.5        
[40] R6_2.5.1            ggridges_0.5.4      lifecycle_1.0.4    
[43] htmlwidgets_1.6.3   MASS_7.3-60         cluster_2.1.5      
[46] pkgconfig_2.0.3     progressr_0.14.0    pillar_1.9.0       
[49] gtable_0.3.4        data.table_1.14.8   glue_1.6.2         
[52] Rcpp_1.0.11         xfun_0.41           tibble_3.2.1       
[55] tidyselect_1.2.0    rstudioapi_0.15.0   knitr_1.45         
[58] igraph_1.5.1        htmltools_0.5.7     compiler_4.3.2     
[61] sp_2.1-2       ```

**Comment by fisheryu2022** [comment]
> Hello,
> 
> Recently I have been getting an error, despite the code working in the past. The input data was not changed in any way and I noticed this error was occurring only after updating to Seurat 5.0.1. In addition, when running on a computer with Seurat 4.3.0 and Seurat Object 4.1.3, the code runs just fine. Below is the error:
> 
> #Converting from h5ad to h5seurat Convert("Mouse_PR_clean.h5ad", dest = "h5seurat", overwrite = TRUE)
> 
> #Loading the h5seurat dataset mouse <- LoadH5Seurat("Mouse_PR_clean.h5seurat", meta.data = FALSE, misc = FALSE, array = "RNA")
> 
> #Standard Pre-processing of Mouse Dataset mouse <- NormalizeData(mouse) mouse <- FindVariableFeatures(mouse, nfeatures = 2000)
> 
> Error in order(hvf.info$vst.variance.standardized, decreasing = TRUE) : argument 1 is not a vector
> 
> To fix this error, I have been trying to degrade the Seurat version following the instructions on the seurat website, and although I converted Seurat and Seurat Object to the required lower versions, the error still persists, which may be because the CRAN dependency packages are still a higher version. Is this a correct interpretation of the error and how is it recommended to fix it? Thanks in advance!

I saw similar problem yesterday. The same coding worked 3 days ago (Seurat 5.0.0), but after updating to Seurat 5.0.1, it stopped working. Then I downgrade to Seurat 4.4.0, SeuratObject 4.1.14, another problem comes - "Error in irlba::irlba(L, nv = n, nu = 0, maxit = iters) : function 'as_cholmod_sparse' not provided by package 'Matrix', which disallows the function of RunUmap. I am using a M1 chip.  Can anyone help?

**Reply by HappinessEricst** [reply]
Hi i had this issue too and after installing this package it worked  (RSpectra)

**Reply by francesbee9** [reply]
Hi, I  also had this issue and had to follow the below to get it working again:
https://stackoverflow.com/questions/69639782/installing-gfortran-on-macbook-with-apple-m1-chip-for-use-in-r

**Comment by yky010894** [comment]
I faced similar problem. And I found that the problem may come from the SeuratObject class Assay version. However, I have to stick into the V5 Seurat so I used the method below:

https://satijalab.org/seurat/articles/seurat5_essential_commands
```
We can also convert (cast) between Assay and Assay5 objects with as().
# convert a v5 assay to a v3 assay
pbmc3k[["RNA3"]] <- as(object = pbmc3k[["RNA"]], Class = "Assay")
# convert a v3 assay to a v5 assay
pbmc3k[["RNA5"]] <- as(object = pbmc3k[["RNA3"]], Class = "Assay5")
```

My code
```
# convert a v3 assay to a v5 assay when you are using version 5 Seurat, but the seurat object you have on hands is not Assay5 
your_object[["RNA"]] <- as(object = your_object[["RNA"]], Class = "Assay5")

```
Hope it could help.

**Reply by yuanG2000** [reply]
> I faced similar problem. And I found that the problem may come from the SeuratObject class Assay version. However, I have to stick into the V5 Seurat so I used the method below:
> 
> https://satijalab.org/seurat/articles/seurat5_essential_commands
> 
> ```
> We can also convert (cast) between Assay and Assay5 objects with as().
> # convert a v5 assay to a v3 assay
> pbmc3k[["RNA3"]] <- as(object = pbmc3k[["RNA"]], Class = "Assay")
> # convert a v3 assay to a v5 assay
> pbmc3k[["RNA5"]] <- as(object = pbmc3k[["RNA3"]], Class = "Assay5")
> ```
> 
> My code
> 
> ```
> # convert a v3 assay to a v5 assay when you are using version 5 Seurat, but the seurat object you have on hands is not Assay5 
> your_object[["RNA"]] <- as(object = your_object[["RNA"]], Class = "Assay5")
> ```
> 
> Hope it could help.

thankyou so much! I solved the problem
